### PR TITLE
fix: delay concluding attempt in pushsync

### PR DESCRIPTION
**Pull request by metacertain**
This PR intends to make forwarder behavior ensure that chunk is successfully written to at least one further peer before concluding attempt has been made

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2016)

### CODE REVIEW

File: pkg/pushsync/pushsync_test.go [hunk 1]
```diff
@@ -954,15 +952,25 @@ func TestPushChunkToClosestSkipFailed(t *testing.T) {
...
+			func(h p2p.HandlerFunc) p2p.HandlerFunc {
+				return func(ctx context.Context, peer p2p.Peer, stream p2p.Stream) error {
+					lock.Lock()
+					defer lock.Unlock()
 
-				return nil
+					if triggerCount < 9 {
+						triggerCount++
+						stream.Close()
+						return errors.New("fmt")
```
Comment (aloknerurkar) [inline pkg/pushsync/pushsync_test.go:964]: Use a better error here.

**Review by aloknerurkar — APPROVED** [review]

File: pkg/pushsync/pushsync.go [hunk 2]
```diff
@@ -470,6 +461,15 @@ func (ps *PushSync) pushPeer(ctx context.Context, peer swarm.Address, ch swarm.C
 		return nil, true, err
 	}
 
+	// if you manage to get a tag, just increment the respective counter
+	t, err := ps.tagger.Get(ch.TagID())
```
Comment (acud) [inline pkg/pushsync/pushsync.go:465]: this is not correct... you should increment the sent count once the chunk is sent.

File: pkg/pushsync/pushsync_test.go [hunk 3]
```diff
@@ -488,7 +488,7 @@ func TestPushChunkToNextClosest(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if ta2.Get(tags.StateSent) != 2 {
```
Comment (acud) [inline pkg/pushsync/pushsync_test.go:491]: this seems a bit wrong no? why is this change needed?

**Review by acud — CHANGES_REQUESTED** [review]
tag increment needs to be reverted to previous state

File: pkg/pushsync/pushsync.go (hunk 2 above)
Comment (metacertain) [reply pkg/pushsync/pushsync.go:465]: ok, it was a misunderstanding

File: pkg/pushsync/pushsync_test.go (hunk 3 above)
Comment (metacertain) [reply pkg/pushsync/pushsync_test.go]: restored as well

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2016)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @acud and @metacertain)

___
*[pkg/pushsync/pushsync_test.go, line 944 at r3](https://reviewable.io/reviews/ethersphere/bee/2016#-MbkH6mRD0zCYWmXfmCO:-MbkH6mSE2NpaEjsWjCw:bg1io4a) ([raw file](https://github.com/ethersphere/bee/blob/9c237ed82b893bce65948a82f256bb357e1d2389/pkg/pushsync/pushsync_test.go#L944)):*
> ```Go
> 
> 	triggerCount := 0
> 	var lock sync.Mutex
> ```

Does this lock suppose to guard just the `triggerCount` variable? If so, I'd recommend to rename it to signal that those two are related.

**Review by esadakar — APPROVED** [review]

**Review by acud — APPROVED** [review]
